reactive/spark.py

```diff
@@ -30,81 +30,77 @@ def report_status():
     hookenv.status_set('active', 'Ready ({})'.format(mode))
 
 
-def install_spark(hadoop, yarn=False):
+def install_spark(hadoop=None):
     spark_master_host = leadership.leader_get('master-fqdn')
-    nns = hadoop.namenodes()
     hosts = {
-        'namenode': nns[0],
         'spark-master': spark_master_host,
     }
 
-    if yarn:
+    if is_state('hadoop.yarn.ready'):
         rms = hadoop.resourcemanagers()
         hosts['resourcemanager'] = rms[0]
 
+    if is_state('hadoop.hdfs.ready'):
+        nns = hadoop.namenodes()
+        hosts['namenode'] = nns[0]
+
     dist = get_dist_config()
     spark = Spark(dist)
     spark.configure(hosts)
 
-@when_any('spark.standalone.installed', 'spark.yarn.installed' )
-@when('config.changed', 'hadoop.hdfs.ready')
-def reconfigure_spark(hadoop):
-    config = hookenv.config()
-    mode = hookenv.config()['spark_execution_mode']
-    hookenv.status_set('maintenance', 
-                       'Changing default execution mode to {}'.format(mode))
-
-    yarn = False
-    if is_state('hadoop.yarn.ready'):
-        yarn = True
 
-    install_spark(hadoop, yarn)
+@when('bigtop.available')
+@when_not('spark.started')
```

Comment:
Will this event re-fire when spark is restarted? If so, is first_install_spark idempotent? (My guess would be not.)